src/managers/A5eEnricherManager.ts

```diff
@@ -0,0 +1,243 @@
+import type { BaseActorA5e } from '../documents/actor/base';
+
+declare namespace A5eEnricherManager {
+  type EnricherTypes = 'check' | 'save';
+}
+
+class A5eEnricherManager {
+  constructor() {
+    this.registerCustomEnrichers();
+  }
+
+  registerCustomEnrichers() {
+    const enricherTypes = ['check', 'save'];
+
+    CONFIG.TextEditor.enrichers.push({
+      // pattern: \[\[\/(?<enricherType>\w+)(?<argString>( +\w+=([\w\d]+|"[\w\d ]+"))*)\]\]
+      // matches: [[/type arg1=val1 arg2=val2 arg3="val 3"]]
+      pattern: new RegExp(`\\[\\[\\/(?<enricherType>${enricherTypes.join('|')})(?<argString>( +\\w+=([\\w\\d]+|"[\\w\\d ]+"))*)\\]\\]`, 'gi'),
+      enricher: this.parseEnricherInput.bind(this)
+    });
+    document.body.addEventListener('click', this.rollEvent);
+  }
+
+  /**
+   * Parse the enriched string and provide the appropriate content.
+   * @param {RegExpMatchArray} match       The regular expression match result.
+   * @param {EnrichmentOptions} options    Options provided to customize text enrichment.
+   * @returns {Promise<HTMLElement|null>}  An HTML element to insert in place of the matched text or
+   *                                       null to indicate that no replacement should be made.
+  */
+  parseEnricherInput(match, options) {
+    const { enricherType, argString } = match.groups;
+
+    const args = this.parseArguments(argString);
+    args.enricherType = enricherType;
+
+    if (enricherType.toLowerCase() === 'check') {
```

Comment:
Better to have enricherType be lowercased once. Since it'll be called multiple times.